PersistentActorJavaDslTest: tapPersistentActor expected akka.persistence.typed.javadsl.PersistentActorJavaDslTest$Increment@4c84057d, found RecoveryPermitGranted

multi-node: https://jenkins.akka.io:8498/blue/rest/organizations/jenkins/pipelines/akka-multi-node/runs/2980/log/?start=0
```
[info] PersistentActorJavaDslTest:
[info] - chainableSideEffectsWithEvents
[info] - transformEvent
[info] - tapPersistentActor *** FAILED ***
[info]   java.lang.AssertionError: assertion failed: expected akka.persistence.typed.javadsl.PersistentActorJavaDslTest$Increment@4c84057d, found RecoveryPermitGranted
[info]   at scala.Predef$.assert(Predef.scala:219)
[info]   at akka.actor.testkit.typed.internal.TestProbeImpl.expectMessage_internal(TestProbeImpl.scala:137)
[info]   at akka.actor.testkit.typed.internal.TestProbeImpl.expectMessage(TestProbeImpl.scala:120)
[info]   at akka.persistence.typed.javadsl.PersistentActorJavaDslTest.tapPersistentActor(PersistentActorJavaDslTest.java:454)
[info]   at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[info]   at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[info]   at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[info]   at java.lang.reflect.Method.invoke(Method.java:497)
[info]   at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
[info]   at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
[info]   at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
[info]   at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
[info]   at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
[info]   at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
[info]   at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
[info]   at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
[info]   at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
[info]   at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
[info]   at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
[info]   at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
[info]   at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:48)
[info]   at org.junit.rules.RunRules.evaluate(RunRules.java:20)
[info]   at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
[info]   at org.junit.runners.Suite.runChild(Suite.java:128)
[info]   at org.junit.runners.Suite.runChild(Suite.java:27)
[info]   at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
[info]   at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
[info]   at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
[info]   at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
[info]   at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
[info]   at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
[info]   at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
[info]   at org.junit.runner.JUnitCore.run(JUnitCore.java:115)
[info]   at org.junit.runner.JUnitCore.run(JUnitCore.java:105)
[info]   at org.junit.runner.JUnitCore.run(JUnitCore.java:94)
[info]   at org.scalatest.junit.JUnitSuiteLike.run(JUnitSuiteLike.scala:242)
[info]   at org.scalatest.junit.JUnitSuiteLike.run$(JUnitSuiteLike.scala:230)
[info]   at org.scalatest.junit.JUnitSuite.run(JUnitSuite.scala:71)
[info]   at org.scalatest.tools.Framework.org$scalatest$tools$Framework$$runSuite(Framework.scala:314)
[info]   at org.scalatest.tools.Framework$ScalaTestTask.execute(Framework.scala:507)
[info]   at sbt.ForkMain$Run.lambda$runTest$1(ForkMain.java:304)
[info]   at sbt.ForkMain$Run$$Lambda$435/1822630059.call(Unknown Source)
[info]   at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[info]   at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
[info]   at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
[info]   at java.lang.Thread.run(Thread.java:745)
```

Another failure in the same test:
https://jenkins.akka.io:8498/blue/rest/organizations/jenkins/pipelines/akka-multi-node-nightly/runs/6438/log/?start=0

```
[info] - stopThenLog *** FAILED ***
[info]   java.lang.AssertionError: assertion failed: timeout (1 second) during expectStop waiting for actor [akka://PersistentActorJavaDslTest/user/$n] to stop
[info]   at scala.Predef$.assert(Predef.scala:219)
[info]   at akka.actor.testkit.typed.internal.TestProbeImpl.expectTerminated(TestProbeImpl.scala:228)
[info]   at akka.actor.testkit.typed.internal.TestProbeImpl.expectTerminated(TestProbeImpl.scala:233)
[info]   at akka.persistence.typed.javadsl.PersistentActorJavaDslTest.stopThenLog(PersistentActorJavaDslTest.java:432)
```

--- a/akka-actor-typed-tests/src/test/java/akka/actor/typed/javadsl/InterceptTest.java
+++ b/akka-actor-typed-tests/src/test/java/akka/actor/typed/javadsl/InterceptTest.java
@@ -1,0 +1,86 @@
+/*
+ * Copyright (C) 2009-2019 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.actor.typed.javadsl;
+
+import akka.actor.testkit.typed.javadsl.TestKitJunitResource;
+import akka.actor.testkit.typed.javadsl.TestProbe;
+import akka.actor.typed.*;
+import akka.testkit.AkkaSpec;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.scalatest.junit.JUnitSuite;
+
+public class InterceptTest extends JUnitSuite {
+
+  @ClassRule
+  public static final TestKitJunitResource testKit = new TestKitJunitResource(AkkaSpec.testConf());
+
+  @Test
+  public void interceptMessage() {
+    final TestProbe<String> interceptProbe = testKit.createTestProbe();
+    BehaviorInterceptor<String, String> interceptor = new BehaviorInterceptor<String, String>() {
+      @Override
+      public Behavior<String> aroundReceive(TypedActorContext<String> ctx, String msg, ReceiveTarget<String> target) {
+        interceptProbe.getRef().tell(msg);
+        return target.apply(ctx, msg);
+      }
+
+      @Override
+      public Behavior<String> aroundSignal(TypedActorContext<String> ctx, Signal signal, SignalTarget<String> target) {
+        return target.apply(ctx, signal);
+      }
+    };
+
+    final TestProbe<String> probe = testKit.createTestProbe();
+    ActorRef<String> ref = testKit.spawn(Behaviors.intercept(interceptor, Behaviors.receiveMessage((String msg) -> {
+      probe.getRef().tell(msg);
+      return Behaviors.same();
+    })));
+    ref.tell("Hello");
+
+    interceptProbe.expectMessage("Hello");
+    probe.expectMessage("Hello");
+  }
+
+  interface Message {}
+  static class A implements Message {}
+  static class B implements Message {}
+
+  @Test
+  public void interceptMessagesSelectively() {
+    final TestProbe<Message> interceptProbe = testKit.createTestProbe();
+    BehaviorInterceptor<Message, Message> interceptor = new BehaviorInterceptor<Message, Message>() {
+
+      @Override
+      public Class<? extends Message> interceptMessageType() {
+        return B.class;
+      }
+
+      @Override
+      public Behavior<Message> aroundReceive(TypedActorContext<Message> ctx, Message msg, ReceiveTarget<Message> target) {
+        interceptProbe.getRef().tell(msg);
+        return target.apply(ctx, msg);
+      }
+
+      @Override
+      public Behavior<Message> aroundSignal(TypedActorContext<Message> ctx, Signal signal, SignalTarget<Message> target) {
+        return target.apply(ctx, signal);
+      }
+    };
+
+    final TestProbe<Message> probe = testKit.createTestProbe();
+    ActorRef<Message> ref = testKit.spawn(Behaviors.intercept(interceptor, Behaviors.receiveMessage((Message msg) -> {
+      probe.getRef().tell(msg);
+      return Behaviors.same();
+    })));
+    ref.tell(new A());
+    ref.tell(new B());
+
+    probe.expectMessageClass(A.class);
+    interceptProbe.expectMessageClass(B.class);
+    probe.expectMessageClass(B.class);
+  }
+
+}

--- a/akka-actor-typed-tests/src/test/scala/akka/actor/typed/InterceptSpec.scala
+++ b/akka-actor-typed-tests/src/test/scala/akka/actor/typed/InterceptSpec.scala
@@ -10,8 +10,8 @@ import akka.testkit.EventFilter
 import akka.actor.testkit.typed.scaladsl.TestProbe
 import akka.actor.typed.scaladsl.Behaviors
 import org.scalatest.WordSpecLike
-import scala.concurrent.duration._
 
+import scala.concurrent.duration._
 import akka.actor.ActorInitializationException
 import akka.actor.testkit.typed.scaladsl.ScalaTestWithActorTestKit
 
@@ -46,7 +46,7 @@ class InterceptSpec extends ScalaTestWithActorTestKit(
     // keeping the instance equality as "isSame" for these
   }
 
-  "Intercept" should {
+  "Intercept" must {
 
     "intercept messages" in {
       val probe = TestProbe[String]()
@@ -198,115 +198,149 @@ class InterceptSpec extends ScalaTestWithActorTestKit(
       innerBehaviorStarted.get should ===(false)
     }
 
-  }
+    "intercept with nested setup" in {
+      val probe = TestProbe[String]()
+      val interceptor = snitchingInterceptor(probe.ref)
 
-  "intercept with nested setup" in {
-    val probe = TestProbe[String]()
-    val interceptor = snitchingInterceptor(probe.ref)
-
-    val ref: ActorRef[String] = spawn(Behaviors.intercept(interceptor)(
-      Behaviors.setup { _ ⇒
-        var count = 0
-        Behaviors.receiveMessage[String] { m ⇒
-          count += 1
-          probe.ref ! s"actual behavior $m-$count"
-          Behaviors.same
-        }
-      }
-    ))
-
-    ref ! "a"
-    probe.expectMessage("before a")
-    probe.expectMessage("actual behavior a-1")
-    probe.expectMessage("after a")
-
-    ref ! "b"
-    probe.expectMessage("before b")
-    probe.expectMessage("actual behavior b-2")
-    probe.expectMessage("after b")
-  }
-
-  "intercept with recursivly setup" in {
-    val probe = TestProbe[String]()
-    val interceptor = snitchingInterceptor(probe.ref)
-
-    def next(count1: Int): Behavior[String] = {
-      Behaviors.intercept(interceptor)(
+      val ref: ActorRef[String] = spawn(Behaviors.intercept(interceptor)(
         Behaviors.setup { _ ⇒
-          var count2 = 0
+          var count = 0
           Behaviors.receiveMessage[String] { m ⇒
-            count2 += 1
-            probe.ref ! s"actual behavior $m-$count1-$count2"
-            next(count1 + 1)
+            count += 1
+            probe.ref ! s"actual behavior $m-$count"
+            Behaviors.same
           }
         }
-      )
+      ))
+
+      ref ! "a"
+      probe.expectMessage("before a")
+      probe.expectMessage("actual behavior a-1")
+      probe.expectMessage("after a")
+
+      ref ! "b"
+      probe.expectMessage("before b")
+      probe.expectMessage("actual behavior b-2")
+      probe.expectMessage("after b")
     }
 
-    val ref: ActorRef[String] = spawn(next(1))
+    "intercept with recursivly setup" in {
+      val probe = TestProbe[String]()
+      val interceptor = snitchingInterceptor(probe.ref)
 
-    ref ! "a"
-    probe.expectMessage("before a")
-    probe.expectMessage("actual behavior a-1-1")
-    probe.expectMessage("after a")
+      def next(count1: Int): Behavior[String] = {
+        Behaviors.intercept(interceptor)(
+          Behaviors.setup { _ ⇒
+            var count2 = 0
+            Behaviors.receiveMessage[String] { m ⇒
+              count2 += 1
+              probe.ref ! s"actual behavior $m-$count1-$count2"
+              next(count1 + 1)
+            }
+          }
+        )
+      }
 
-    ref ! "b"
-    probe.expectMessage("before b")
-    probe.expectMessage("actual behavior b-2-1")
-    probe.expectMessage("after b")
+      val ref: ActorRef[String] = spawn(next(1))
 
-    ref ! "c"
-    probe.expectMessage("before c")
-    probe.expectMessage("actual behavior c-3-1")
-    probe.expectMessage("after c")
-  }
+      ref ! "a"
+      probe.expectMessage("before a")
+      probe.expectMessage("actual behavior a-1-1")
+      probe.expectMessage("after a")
 
-  "not allow intercept setup(same)" in {
-    val probe = TestProbe[String]()
-    val interceptor = snitchingInterceptor(probe.ref)
+      ref ! "b"
+      probe.expectMessage("before b")
+      probe.expectMessage("actual behavior b-2-1")
+      probe.expectMessage("after b")
 
-    EventFilter[ActorInitializationException](occurrences = 1).intercept {
-      val ref = spawn(Behaviors.intercept(interceptor)(
-        Behaviors.setup[String] { _ ⇒ Behaviors.same[String] }))
+      ref ! "c"
+      probe.expectMessage("before c")
+      probe.expectMessage("actual behavior c-3-1")
+      probe.expectMessage("after c")
+    }
+
+    "not allow intercept setup(same)" in {
+      val probe = TestProbe[String]()
+      val interceptor = snitchingInterceptor(probe.ref)
+
+      EventFilter[ActorInitializationException](occurrences = 1).intercept {
+        val ref = spawn(Behaviors.intercept(interceptor)(
+          Behaviors.setup[String] { _ ⇒ Behaviors.same[String] }))
+        probe.expectTerminated(ref, probe.remainingOrDefault)
+      }
+
+    }
+
+    "be useful for implementing PoisonPill" in {
+
+      def inner(count: Int): Behavior[Msg] = Behaviors.receiveMessage {
+        case Msg(hello, replyTo) ⇒
+          replyTo ! s"$hello-$count"
+          inner(count + 1)
+      }
+
+      val poisonInterceptor = new BehaviorInterceptor[Any, Msg] {
+        override def aroundReceive(context: TypedActorContext[Any], message: Any, target: ReceiveTarget[Msg]): Behavior[Msg] =
+          message match {
+            case MyPoisonPill ⇒ Behaviors.stopped
+            case m: Msg       ⇒ target(context, m)
+            case _            ⇒ Behaviors.unhandled
+          }
+
+        override def aroundSignal(context: TypedActorContext[Any], signal: Signal, target: SignalTarget[Msg]): Behavior[Msg] =
+          target.apply(context, signal)
+
+      }
+
+      val decorated: Behavior[Msg] =
+        Behaviors.intercept(poisonInterceptor)(inner(0)).narrow
+
+      val ref = spawn(decorated)
+      val probe = TestProbe[String]()
+      ref ! Msg("hello", probe.ref)
+      probe.expectMessage("hello-0")
+      ref ! Msg("hello", probe.ref)
+      probe.expectMessage("hello-1")
+
+      ref.unsafeUpcast[Any] ! MyPoisonPill
+
       probe.expectTerminated(ref, probe.remainingOrDefault)
     }
 
-  }
+    "be able to intercept a subset of the messages" in {
+      trait Message
+      class A extends Message
+      class B extends Message
 
-  "be useful for implementing PoisonPill" in {
+      val interceptProbe = TestProbe[Message]()
 
-    def inner(count: Int): Behavior[Msg] = Behaviors.receiveMessage {
-      case Msg(hello, replyTo) ⇒
-        replyTo ! s"$hello-$count"
-        inner(count + 1)
-    }
+      val partialInterceptor: BehaviorInterceptor[Message, Message] = new BehaviorInterceptor[Message, Message] {
 
-    val poisonInterceptor = new BehaviorInterceptor[Any, Msg] {
-      override def aroundReceive(context: TypedActorContext[Any], message: Any, target: ReceiveTarget[Msg]): Behavior[Msg] =
-        message match {
-          case MyPoisonPill ⇒ Behaviors.stopped
-          case m: Msg       ⇒ target(context, m)
-          case _            ⇒ Behaviors.unhandled
+        override def interceptMessageType = classOf[B]
+
+        override def aroundReceive(ctx: TypedActorContext[Message], msg: Message, target: ReceiveTarget[Message]): Behavior[Message] = {
+          interceptProbe.ref ! msg
+          target(ctx, msg)
         }
 
-      override def aroundSignal(context: TypedActorContext[Any], signal: Signal, target: SignalTarget[Msg]): Behavior[Msg] =
-        target.apply(context, signal)
+        override def aroundSignal(ctx: TypedActorContext[Message], signal: Signal, target: SignalTarget[Message]): Behavior[Message] =
+          target(ctx, signal)
+      }
 
+      val probe = TestProbe[Message]()
+      val ref = spawn(Behaviors.intercept(partialInterceptor)(Behaviors.receiveMessage { msg ⇒
+        probe.ref ! msg
+        Behaviors.same
+      }))
+
+      ref ! new A
+      ref ! new B
+
+      probe.expectMessageType[A]
+      interceptProbe.expectMessageType[B]
+      probe.expectMessageType[B]
     }
 
-    val decorated: Behavior[Msg] =
-      Behaviors.intercept(poisonInterceptor)(inner(0)).narrow
-
-    val ref = spawn(decorated)
-    val probe = TestProbe[String]()
-    ref ! Msg("hello", probe.ref)
-    probe.expectMessage("hello-0")
-    ref ! Msg("hello", probe.ref)
-    probe.expectMessage("hello-1")
-
-    ref.unsafeUpcast[Any] ! MyPoisonPill
-
-    probe.expectTerminated(ref, probe.remainingOrDefault)
   }
 
 }

--- a/akka-actor-typed/src/main/scala/akka/actor/typed/BehaviorInterceptor.scala
+++ b/akka-actor-typed/src/main/scala/akka/actor/typed/BehaviorInterceptor.scala
@@ -18,11 +18,15 @@ abstract class BehaviorInterceptor[O, I] {
   import BehaviorInterceptor._
 
   /**
-   * Allows for bypassing the interceptor for certain messages. Useful if the official protocol and the actual
+   * Allows for applying the interceptor only to certain message types. Useful if the official protocol and the actual
    * protocol of an actor causes problems, for example class cast exceptions for a message not of type `O` that
-   * the actor still knows how to deal with.
+   * the actor still knows how to deal with. Note that this is only possible to use when `O` and `I` are the same type.
+   *
+   * @return A subtype of `O` that should be intercepted or `null` to intercept all `O`s.
+   *         Subtypes of `O` matching this are passed directly to the inner behavior without interception.
    */
-  def applyFor(message: Any): Boolean = true
+  // null for all to avoid having to deal with class tag/explicit class in the default case of no filter
+  def interceptMessageType: Class[_ <: O] = null
 
   /**
    * Override to intercept actor startup. To trigger startup of

--- a/akka-actor-typed/src/main/scala/akka/actor/typed/BehaviorInterceptor.scala
+++ b/akka-actor-typed/src/main/scala/akka/actor/typed/BehaviorInterceptor.scala
@@ -18,6 +18,13 @@ abstract class BehaviorInterceptor[O, I] {
   import BehaviorInterceptor._
 
   /**
+   * Allows for bypassing the interceptor for certain messages. Useful if the official protocol and the actual
+   * protocol of an actor causes problems, for example class cast exceptions for a message not of type `O` that
+   * the actor still knows how to deal with.
+   */
+  def applyFor(message: Any): Boolean = true
+
+  /**
    * Override to intercept actor startup. To trigger startup of
    * the next behavior in the stack, call `target.start()`.
    * @return The returned behavior will be the "started" behavior of the actor used to accept

--- a/akka-actor-typed/src/main/scala/akka/actor/typed/internal/InterceptorImpl.scala
+++ b/akka-actor-typed/src/main/scala/akka/actor/typed/internal/InterceptorImpl.scala
@@ -7,7 +7,7 @@ package akka.actor.typed.internal
 import akka.actor.typed
 import akka.actor.typed.Behavior.{ SameBehavior, UnhandledBehavior }
 import akka.actor.typed.internal.TimerSchedulerImpl.TimerMsg
-import akka.actor.typed.{ TypedActorContext, ActorRef, Behavior, BehaviorInterceptor, ExtensibleBehavior, PreRestart, Signal }
+import akka.actor.typed.{ ActorRef, Behavior, BehaviorInterceptor, ExtensibleBehavior, PreRestart, Signal, TypedActorContext }
 import akka.annotation.InternalApi
 import akka.util.LineNumbers
 
@@ -67,8 +67,12 @@ private[akka] final class InterceptorImpl[O, I](val interceptor: BehaviorInterce
     new InterceptorImpl(interceptor, newNested)
 
   override def receive(ctx: typed.TypedActorContext[O], msg: O): Behavior[O] = {
-    val interceptedResult = interceptor.aroundReceive(ctx, msg, receiveTarget)
-    deduplicate(interceptedResult, ctx)
+    val result =
+      if (interceptor.applyFor(msg))
+        interceptor.aroundReceive(ctx, msg, receiveTarget)
+      else
+        receiveTarget.apply(ctx, msg.asInstanceOf[I])
+    deduplicate(result, ctx)
   }
 
   override def receiveSignal(ctx: typed.TypedActorContext[O], signal: Signal): Behavior[O] = {

--- a/akka-actor-typed/src/main/scala/akka/actor/typed/internal/InterceptorImpl.scala
+++ b/akka-actor-typed/src/main/scala/akka/actor/typed/internal/InterceptorImpl.scala
@@ -67,8 +67,9 @@ private[akka] final class InterceptorImpl[O, I](val interceptor: BehaviorInterce
     new InterceptorImpl(interceptor, newNested)
 
   override def receive(ctx: typed.TypedActorContext[O], msg: O): Behavior[O] = {
+    val interceptMessageType = interceptor.interceptMessageType
     val result =
-      if (interceptor.applyFor(msg))
+      if (interceptMessageType == null || interceptMessageType.isAssignableFrom(msg.getClass))
         interceptor.aroundReceive(ctx, msg, receiveTarget)
       else
         receiveTarget.apply(ctx, msg.asInstanceOf[I])

--- a/akka-persistence-typed/src/test/java/akka/persistence/typed/javadsl/PersistentActorJavaDslTest.java
+++ b/akka-persistence-typed/src/test/java/akka/persistence/typed/javadsl/PersistentActorJavaDslTest.java
@@ -451,6 +451,12 @@ public class PersistentActorJavaDslTest extends JUnitSuite {
     TestProbe<Object> interceptProbe = testKit.createTestProbe();
     TestProbe<Signal> signalProbe = testKit.createTestProbe();
     BehaviorInterceptor<Command, Command> tap = new BehaviorInterceptor<Command, Command>() {
+
+      @Override
+      public boolean applyFor(Object msg) {
+        return msg instanceof Command;
+      }
+
       @Override
       public Behavior<Command> aroundReceive(TypedActorContext<Command> ctx, Command msg, ReceiveTarget<Command> target) {
         interceptProbe.ref().tell(msg);

--- a/akka-persistence-typed/src/test/java/akka/persistence/typed/javadsl/PersistentActorJavaDslTest.java
+++ b/akka-persistence-typed/src/test/java/akka/persistence/typed/javadsl/PersistentActorJavaDslTest.java
@@ -453,8 +453,8 @@ public class PersistentActorJavaDslTest extends JUnitSuite {
     BehaviorInterceptor<Command, Command> tap = new BehaviorInterceptor<Command, Command>() {
 
       @Override
-      public boolean applyFor(Object msg) {
-        return msg instanceof Command;
+      public Class<Command> interceptMessageType() {
+        return Command.class;
       }
 
       @Override

